commerce/domain/pricing/service/classes/PricingServiceSample.cls

```diff
@@ -139,13 +138,18 @@ public class PricingServiceSample extends commercestorepricing.PricingService {
       txnResponse.getTotalProductAmount() + txnResponse.getTotalAdjustmentAmount()
     );
 
+    /**
+     * Override success/failure of a product easily by adding an error message to the product.
+     * Here is a sample code demonstrating how to set error messages for products.
+     * We are failing the first product in the response.
+     * 
     if (!txnItemCollection.isEmpty()) {
-      // Override success/failure of a product easily by adding an error message to the product. Here
-      // we are failing the first product in the response.
       String customErrorMessage = 'We no longer sell this particular product.';
       String localizedErrorMessage = 'Wir verkaufen dieses spezielle Produkt nicht mehr.';
       txnItemCollection.get(0).setError(customErrorMessage, localizedErrorMessage);
     }
+	  */
+
```

Comment:
@adarshjain-sf Not sure if leaving commented code would be the best practice here(Assuming this code sample is part of public doc). Better to remove the code if we don't want it to be here.
In case this is a sample use case that we want to document,  maybe we can add a new class all together to demonstrate how to override success/failure of a product.

Reply:
@anwebanSalesforce As discussed, this code is mainly to guide users using pricing extension on how they can set the custom errors. So we don't want to remove it from the code.
But we have modified the comment that using the given sample code directly would cause cart operations to fail.